### PR TITLE
feat(developer): New LDML Keyboard Project form 🌱

### LINKED ISSUE #10135: feat(developer): need a way to create an LDML keyboard project
### Is your feature request related to a problem? Please describe.

Keyman Developer needs a LDML Keyboard Project template in New Project

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Related issues

_No response_

### Keyman apps

- [ ] Keyman for Android
- [ ] Keyman for iPhone and iPad
- [ ] Keyman for Linux
- [ ] Keyman for macOS
- [ ] Keyman for Windows
- [X] Keyman Developer
- [ ] KeymanWeb
- [ ] Other - give details at bottom of form

### Keyman version

_No response_

### Operating system

_No response_

### Device

_No response_

### Target application

_No response_

### Browser

_No response_

### Keyboard name

_No response_

### Keyboard version

_No response_

### Language name

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #10135.

Adds a new LDML Keyboard Project template and form, and allows instantiation from kmconvert command-line and from the New Project dialog in Keyman Developer IDE.

# User Testing

**TEST_NEW_LDML_PROJECT:** Using the New Project dialog, create a new LDML Keyboard. Verify that the dialog operates as you would expect. For the 'language codes' field, you can use any valid BCP-47 tag, such as 'en' or 'ta'.

Note: an LDML keyboard is a .xml file, not a .kmn file. Given a keyboard project my_keyboard, the files that are created should be:

- my_keyboard.kpj
- source/my_keyboard.kps
- source/my_keyboard.xml
- source/welcome.htm
- source/readme.htm
- README.md
- LICENSE.md
- HISTORY.md

### CODE REVIEW

**Comment by mcdurdin** [comment]
Returned to draft to improve the template keyboard formatting

**Review by srl295 — COMMENTED** [review]
looks OK as far as I can understand it, would be curious what the template xml looks like

**Comment by mcdurdin** [comment]
Template XML is currently like this:

```xml
<?xml version="1.0"?>
<!DOCTYPE keyboard3 SYSTEM "ldmlKeyboard3.dtd">
<keyboard3 locale="en" conformsTo="techpreview">
  <info author="Author" name="Sample"/>
  <version number="1.0.0"/>
  <keys>
    <key id="a" output="Hello World!"/>
    <key id="grave" output="`"/>
    <key id="hyphen" output="-"/>
    <key id="equal" output="="/>
    <key id="open-bracket" output="["/>
    <key id="close-bracket" output="]"/>
    <key id="backslash" output="\"/>
    <key id="semi-colon" output=";"/>
    <key id="quote" output="'"/>
    <key id="comma" output=","/>
    <key id="period" output="."/>
    <key id="slash" output="/"/>
  </keys>
  <layers formId="us">
    <layer modifiers="none">
      <row keys="grave 1 2 3 4 5 6 7 8 9 0 hyphen equal"/>
      <row keys="q w e r t y u i o p open-bracket close-bracket backslash"/>
      <row keys="a s d f g h j k l semi-colon quote"/>
      <row keys="z x c v b n m comma period slash"/>
      <row keys="space"/>
    </layer>
  </layers>
</keyboard3>
```

`Hello World!` is emitted by pressing <kbd>a</kbd>, as an example.

**Comment by bharanidharanj** [comment]
# Test Results

* **TEST_NEW_LDML_PROJECT (PASSED):** Tested with the attached PR build (Keyman Developer 17.0.247-alpha-test-10436) on Windows 11 OS (VM) and here is my observation:  1. Created a new LDML keyboard project then created a new LDML keyboard. Added the language code "ta" while creating the project. Able to see the above-mentioned files in the project view. 

..Projects-Keyboards view

    

..Project-Packaging view

  

..Project-Distribution view

File: developer/src/tike/project/Keyman.Developer.UI.Project.UfrmNewLDMLKeyboardProjectParameters.pas [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+begin
+  if Validate then
+    ModalResult := mrOk;
+end;
+
+procedure TfrmNewLDMLKeyboardProjectParameters.editAuthorChange(Sender: TObject);
+begin
+  EnableControls;
+  if not editCopyright.Modified then
+    editCopyright.Text := 'Copyright '+Char($00A9 {copyright})+' '+Author;
```
Comment (srl295) [inline developer/src/tike/project/Keyman.Developer.UI.Project.UfrmNewLDMLKeyboardProjectParameters.pas:210]: seems like some functions could help here for the (c) stuff

**Review by srl295 — APPROVED** [review]

**Pull request merged**